src/mqt/problemsolver/partialcompiler/evaluator.py

```diff
@@ -0,0 +1,82 @@
+from time import time
+
+from mqt.problemsolver.partialcompiler.Partial_QAOA import Partial_QAOA
+
+
+def evaluate_QAOA(num_qubits: int = 4, repetitions: int = 3) -> tuple[float, float, float, float]:
+    """
+    Evaluate the performance of the different partial compilation methods.
+    """
+
+    QAOA_problem_instance = Partial_QAOA(num_qubits=num_qubits, repetitions=repetitions)
+
+    """
+    Partial Compilation Method 1: (faulty because mapping of offline edges most likely changes qubit layout)
+    1. Get all partially compiled circuits (without online edges) using QuantumCircuit.compose()
+    2. Compile online edges
+    3. Compose all circuits
+    """
```

Comment:
I like the idea, but I believe this could be really hard to get to work properly.

Reply:
I see that point and will work on that in the next days.